src/Commands/DrushCommands.php

```diff
@@ -47,6 +50,16 @@ protected function io()
     }
 
     /**
+     * Overrides same method in ConfigAwareTrait in order to provide a
+     * DrushConfig as return type. Helps with IDE completion.
+     *
+     * @return \Drush\Config\DrushConfig
```

Comment:
Pushed a commit which adds ConfigAware to DrushCommands. Everything works, except autocomplete! I feel like I'm learning via self flaggelation. 

Is there anything more I should be doing in in DrushCommands::getConfig or elsewhere.

Reply:
Maybe the IDE does not agree that you can override trait methods in classes (even though php itself seems to be of the mind that this is okay).  Do the use-trait-with-renamed-method trick, as if you were overriding one trait with another (move the parent getConfig out of the way) and see if that helps.

I'd say this is an IDE bug, but we should work around it if possible.